common/serial/arch.h

```diff
@@ -0,0 +1,20 @@
+#ifndef ARCH__H
+#define ARCH__H
+
+#include <bf_stdint.h>
+
+class arch_handler {
+
+public:
+	arch_handler() {}
+	virtual ~arch_handler() {}
+
+	virtual void port_write(uint16_t port, uint8_t value) = 0;
+	virtual void port_write(uint16_t port, uint16_t value) = 0;
```

Comment:
since c++ sucks by the fact that it could do type conversions for you.... I would not use function overloading for this. that is..... make a port_write8 and port_write16